### ISSUE TITLE
versions with tensorflow 2 or pytorch

### ISSUE THREAD

**Issue by linminhtoo** [GIST-CSBL/HoTS#2]
dear authors,

I'm an AI engineer who chanced across your work and would really like to try it out and play around with it.

first off, great work on this and it is a very interesting approach to predict binding site as an output rather than use it as an input. having worked on computer vision problems for the last 10 months, I find your application of concepts from YOLO very satisfying. 

I wish to try your work on different datasets on my laptop. unfortunately, it has RTX 3080 GPU and cannot support tensorflow 1, which is too old to have the compatible CUDA support.  for reference, my CUDA version is 11.6 
please correct me if I am wrong.

here are the questions I have:
1) what kind of hardware was used for the experiments in your work?
2) as I am not familiar with tensorflow, I wish to ask if your repo will also work with tensorflow 2?
3) by any chance, do you intend to or already have a PyTorch version? as many of us are more familiar with PyTorch, this would be extremely helpful. plus, PyTorch has much better backward compatibility behavior than tensorflow (ensuring that people can continue to reproduce & apply this work even when PyTorch gets updated to future versions)

I highly appreciate any response from you. I had reached out to Ingoo on LinkedIn but figured that was not the best channel for this discussion. 
Best, Min Htoo from Singapore

**Comment by dlsrnsi** [comment]
Dear Lin Min Htoo,

1. We used Geforce Titan X for training, validation, and test.
2. We didn't try, but I guess it won't work.
3. Sorry, we don't have PyTorch version also.

However, it works with Colab.
Before learning the trained model, you need to downgrade some packages.
Although it gives many warnings, you can ignore it (it does not affect any prediction results)
```
%tensorflow_version 1.x

!wget -c https://repo.continuum.io/miniconda/Miniconda3-py37_4.8.3-Linux-x86_64.sh
!chmod +x Miniconda3-py37_4.8.3-Linux-x86_64.sh
!bash ./Miniconda3-py37_4.8.3-Linux-x86_64.sh -b -f -p /usr/local
!conda install -q -y -c conda-forge python=3.7 rdkit==2020.09.2

!rm /usr/local/lib/python3.7/dist-packages/h5py-3.1.0.dist-info/ -r
!rm /usr/local/lib/python3.7/dist-packages/h5py/ -r
!rm /usr/local/lib/python3.7/dist-packages/h5py.libs/ -r
!rm /usr/local/lib/python3.7/dist-packages/keras* -r

!pip install keras==2.2.4  --force-reinstall
!pip install h5py==2.10.0 --force-reinstall

import tensorflow as tf
import keras

print(tf.__version__, keras.__version__)
```
Would you try this code?

We're planning to this old-style code to torch code, but I'm not sure when it will be finished.

If you have any issues on running on colab, please comment me more.

Thank you.

Sincerely,

Ingoo Lee

**Comment by dlsrnsi** [comment]
I updated code to tf2. But I didn't update model yet. Soon I will update model too. Thank you for your interest to my model!

**Comment by dlsrnsi** [comment]
Dear @linminhtoo 

Sorry for being late,

I updated the code and jupyter notebook

Also, I uploaded the models, and made the tf1 branch.

For prediction, I recommend using tf1-version.

tf2 version used different normalization because of technical use.

[https://www.kaggle.com/questions-and-answers/188767](https://www.kaggle.com/questions-and-answers/188767)

Thank you for your interest in my model 

Sincerely, 

Ingoo Lee
